app/furniture/marketplace/stripe_events_controller.rb

```diff
@@ -24,7 +24,8 @@ def create
         order.events.create(description: "Payment Received")
 
         if marketplace.square_order_notifications_enabled?
-          order.send_to_square_seller_dashboard
+          square_order = Marketplace::SquareOrder.new(order)
```

Comment:
This controller action is getting nasty; and that's 99.999% my fault. To help reduce the sprawl, would you be available to encapsulate the square stuff a bit more tightly? For example `marketplace.square_connection&.send_order(order)`. 

We could make `marketplace.square_connection` return `nil` in cases where Square Order Notifications are not enabled and a tiny little adapter class (say `Marketplace::SquareConnection`) when there is; eliminating the need for an `if` in the controller and shaving off a few lines.

Reply:
Also, heads up but you may have noticed this; because this file opens with:

```
class Marketplace

end
```

Constant lookups will *start* by looking for constants within the `Marketplace` namespace. So `Marketplace::SquareOrder` will mostly likely resolve as `Marketplace::Marketplace::SquareOrder`. I don't think it will ever look for `Marketplace::SquareOrder` so it will probably raise a constant undefined error.

You can double check this with `rails zeitwerk:check` which will parse the constant lookup for everything in the Rails autoload path to make sure it's all wired in correctly.

Personally, I like how the Rails constant lookup means changing the namespace lets you "switch" the objects that will resolve; as it feels very smalltalk, but I also find it infuriating due to how inconsistent and opaque it is. Zeitwerk helped a lot but there are still moments where it's just facepalms into a door frame.

Reply:
Thanks for the tips about constant lookups. I was even reading through https://noelrappin.com/blog/2023/10/better-know-a-ruby-thing-22-constants/ this week to get a better handle on this aspect of Ruby.

Reply:
> eliminating the need for an if in the controller and shaving off a few lines.
Yes lets do this. I reminded of a Sandi Metz talk about programming before conditionals and how generally one should consider OO encapsulation vs this type of imperative programming. Good to be reminded!